### PR TITLE
Backport triv pr16

### PR DESCRIPTION
**Pull request by PastaPastaPasta**

### CODE REVIEW

File: depends/packages/qt.mk [hunk 1]
```diff
@@ -179,10 +179,7 @@ define $(package)_stage_cmds
   $(MAKE) -C src INSTALL_ROOT=$($(package)_staging_dir) $(addsuffix -install_subtargets,$(addprefix sub-,$($(package)_qt_libs))) && cd .. && \
   $(MAKE) -C qttools/src/linguist/lrelease INSTALL_ROOT=$($(package)_staging_dir) install_target && \
   $(MAKE) -C qttools/src/linguist/lupdate INSTALL_ROOT=$($(package)_staging_dir) install_target && \
-  $(MAKE) -C qttranslations INSTALL_ROOT=$($(package)_staging_dir) install_subtargets && \
-  if `test -f qtbase/src/plugins/platforms/xcb/xcb-static/libxcb-static.a`; then \
-    cp qtbase/src/plugins/platforms/xcb/xcb-static/libxcb-static.a $($(package)_staging_prefix_dir)/lib; \
-  fi
+  $(MAKE) -C qttranslations INSTALL_ROOT=$($(package)_staging_dir) install_subtargets
```
Comment (PastaPastaPasta) [inline depends/packages/qt.mk:183]: 20650 is probably dependant on another backport...

File: test/functional/feature_anchors.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env python3
```
Comment (PastaPastaPasta) [inline test/functional/feature_anchors.py:1]: drop 21338, I don't think we have anchors.dat and assosiated logic

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: src/util/system.h [hunk 3]
```diff
@@ -90,7 +90,7 @@ bool FileCommit(FILE *file);
 bool TruncateFile(FILE *file, unsigned int length);
 int RaiseFileDescriptorLimit(int nMinFD);
 void AllocateFileRange(FILE *file, unsigned int offset, unsigned int length);
-bool RenameOver(fs::path src, fs::path dest);
+[[nodiscard]] bool RenameOver(fs::path src, fs::path dest);
```
Comment (UdjinM6) [inline src/util/system.h:93]: 20519: need 17044 to stop complains about `RenameOver` in `init.cpp`
```
init.cpp:872:13: warning: ignoring return value of function declared with 'nodiscard' attribute [-Wunused-result]
            RenameOver(pathBootstrap, pathBootstrapOld);
            ^~~~~~~~~~ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Comment by PastaPastaPasta** [comment]
Done

**Review by UdjinM6 — APPROVED** [review]
utACK

**Pull request merged**